# Исследовательский анализ по мониторингу цен аптечных сетей. Тестовая работа на трудоустройство.

Перед тем, как приступим к обработке и непосредственно исследованиям, выгрузим все библиотеки и откроем датасет, сохранив в переменной df. Изучим столбцы таблицы и их содержимое.

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats as st
import seaborn as sns
import matplotlib.pyplot as plt
import datetime as dt
from google.colab import drive
drive.mount('/content/gdrive')
df = pd.read_csv('gdrive/My Drive/Test_work.csv')
df

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


Товар (офф. Наименование)  Цена  \
0                                   Жидкость для рук 80мл   NaN   
1                   Мильгамма р-р д/в/м введ. 2мл №5 амп.   NaN   
2       Назол Бэби капли наз. 0.125% 10мл д/младенцев ...   NaN   
3              Тантум Прополис пастилки №15 лес.ягоды БАД   NaN   
4                   Нозефрин спрей наз. 50мкг/доза 120доз   NaN   
...                                                   ...   ...   
104415  Дискус композитум р-р д/в/м вв гомеоп. 2,2мл №...   NaN   
104416  Саксенда р-р д/подк.введ. 6 мг/мл 3мл №5 шприц...   NaN   
104417  Саксенда р-р д/подк.введ. 6 мг/мл 3мл №5 шприц...   NaN   
104418  Саксенда р-р д/подк.введ. 6 мг/мл 3мл №5 шприц...   NaN   
104419  Саксенда р-р д/подк.введ. 6 мг/мл 3мл №5 шприц...   NaN   

                 Регион                  Федеральный округ  \
0                Казань      Приволжский федеральный округ   
1                   Уфа      Приволжский федеральный округ   
2              Белгород      Центральный федеральный округ   
3                  Тула      Центральный федеральный округ   
4                Казань      Приволжский федеральный округ   
...                 ...                                ...   
104415         Белгород      Центральный федеральный округ   
104416  Санкт-Петербург  Северо-западный федеральный округ   
104417         Владимир      Центральный федеральный округ   
104418          Воронеж      Центральный федеральный округ   
104419             Тула      Центральный федеральный округ   

                                    Наименование на сайте Цена на сайте  \
0       ЖИДКОСТЬ ДЛЯ РУК АММИАК+ГЛИЦЕРОЛ+ЭТАНОЛ 80МЛ Р...          0,00   
1                            МИЛЬГАММА 2МЛ N5 АМП Р-Р В/М          0,00   
2             НАЗОЛ БЭБИ 0,125% 10МЛ ФЛАК/КАП КАПЛИ НАЗАЛ          0,00   
3       ТАНТУМ ПРОПОЛИС СО ВКУСОМ ЛЕСНЫХ ЯГОД N15 ПАСТ...          0,00   
4       НОЗЕФРИН 50МКГ/ДОЗА 18,0 СПРЕЙ НАЗАЛЬН ДОЗИР 1...          0,00   
...                                                   ...           ...   
104415  ДИСКУС КОМПОЗИТУМ 2,2МЛ N100 АМП Р-Р В/М ГОМЕОПАТ      22159,80   
104416                 Саксенда 6мг/мл шприц-ручка 3мл №5      23620,00   
104417                 Саксенда 6мг/мл шприц-ручка 3мл №5      24010,00   
104418  САКСЕНДА р-р д/инъекц. 6 мг/мл картридж 3 мл №...      24620,00   
104419  САКСЕНДА р-р д/инъекц. 6 мг/мл картридж 3 мл №...      24620,00   

                Сайт Дата выкачки Время выкачки  \
0          apteka.ru   12.01.2022         00:39   
1          apteka.ru   12.01.2022         00:39   
2          apteka.ru   12.01.2022         00:40   
3          apteka.ru   12.01.2022         00:40   
4          apteka.ru   12.01.2022         00:40   
...              ...          ...           ...   
104415     apteka.ru   12.01.2022         02:17   
104416  stolichki.ru   12.01.2022         03:34   
104417  stolichki.ru   12.01.2022         03:50   
104418   maksavit.ru   12.01.2022         03:33   
104419   maksavit.ru   12.01.2022         04:05   

                                                   Ссылка  
0       https://apteka.ru/product/zhidkost-dlya-ruk-am...  
1       https://apteka.ru/product/milgamma-2ml-n5-amp-...  
2       https://apteka.ru/product/nazol-bebi-0125-10ml...  
3       https://apteka.ru/product/tantum-propolis-so-v...  
4       https://apteka.ru/product/nozefrin-50mkgdoza-1...  
...                                                   ...  
104415  https://apteka.ru/product/diskus-kompozitum-22...  
104416  https://stolichki.ru/drugs/saksenda-6mg-ml-shp...  
104417  https://stolichki.ru/drugs/saksenda-6mg-ml-shp...  
104418                https://maksavit.ru/catalog/119499/  
104419                https://maksavit.ru/catalog/119499/  

[104420 rows x 10 columns]

Рассмотрим, какие столбцы содержат определенные типы.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104420 entries, 0 to 104419
Data columns (total 10 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   Товар (офф. Наименование)  104420 non-null  object 
 1   Цена                       0 non-null       float64
 2   Регион                     104420 non-null  object 
 3   Федеральный округ          104420 non-null  object 
 4   Наименование на сайте      104420 non-null  object 
 5   Цена на сайте              104420 non-null  object 
 6   Сайт                       104420 non-null  object 
 7   Дата выкачки               104420 non-null  object 
 8   Время выкачки              104420 non-null  object 
 9   Ссылка                     104420 non-null  object 
dtypes: float64(1), object(9)
memory usage: 8.0+ MB


Почти все столбцы имеют тип 'object' и только столбец Цена имеет вещественный тип, но он пуст.

Проверим, сколько пропущенных значений в таблице

In [ ]:
df.isna().sum()

Товар (офф. Наименование)         0
Цена                         104420
Регион                            0
Федеральный округ                 0
Наименование на сайте             0
Цена на сайте                     0
Сайт                              0
Дата выкачки                      0
Время выкачки                     0
Ссылка                            0
dtype: int64

Вывод: нет данных по столбцу "Цена". По остальным столбцам пропусков нет.

In [ ]:
df['Цена'].unique()

array([nan])

Есть ли дубликаты?

In [ ]:
df.duplicated().sum()

0

Дубликатов нет

Преобразуем столбец "Цена на сайте" в вещественный тип, предварительно заменив запятые на точки.

In [ ]:
df['Цена на сайте'] = df['Цена на сайте'].str.replace(',', '.')

In [ ]:
df['Цена на сайте'].astype(float)

0             0.0
1             0.0
2             0.0
3             0.0
4             0.0
           ...   
104415    22159.8
104416    23620.0
104417    24010.0
104418    24620.0
104419    24620.0
Name: Цена на сайте, Length: 104420, dtype: float64

Объединим столбцы "Дата выкачки" и "Время выкачки" для удобства расчета и приведем к datetime.

In [ ]:
import datetime
df['Время'] = pd.to_datetime(df['Дата выкачки'].astype(str) + ' ' + df['Время выкачки'].astype(str))
df['Время']

0        2022-12-01 00:39:00
1        2022-12-01 00:39:00
2        2022-12-01 00:40:00
3        2022-12-01 00:40:00
4        2022-12-01 00:40:00
                 ...        
104415   2022-12-01 02:17:00
104416   2022-12-01 03:34:00
104417   2022-12-01 03:50:00
104418   2022-12-01 03:33:00
104419   2022-12-01 04:05:00
Name: Время, Length: 104420, dtype: datetime64[ns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104420 entries, 0 to 104419
Data columns (total 11 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   Товар (офф. Наименование)  104420 non-null  object        
 1   Цена                       0 non-null       float64       
 2   Регион                     104420 non-null  object        
 3   Федеральный округ          104420 non-null  object        
 4   Наименование на сайте      104420 non-null  object        
 5   Цена на сайте              104420 non-null  object        
 6   Сайт                       104420 non-null  object        
 7   Дата выкачки               104420 non-null  object        
 8   Время выкачки              104420 non-null  object        
 9   Ссылка                     104420 non-null  object        
 10  Время                      104420 non-null  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), object(9)
memory u

Посмотрим, какие даты есть в датасете.

In [ ]:
df["Дата выкачки"].unique()

array(['12.01.2022'], dtype=object)

Данные есть только за 12 января 2022 года.

Рассмотрим столбец "Сайт", все ли аптеки, указанные в договоре есть в нашем датасете?

In [ ]:
df['Сайт'].unique()

array(['apteka.ru', 'aptekiplus.ru', 'planetazdorovo.ru', 'stolichki.ru',
       'acmespb.ru', 'maksavit.ru'], dtype=object)

Как видим, в нашей таблице нет таких сетей как aptekamos.ru, ozerki.ru, po-aptekam.ru и rigla.ru. А по условию договора, нам необходимо исследовать цены и в этих сетях аптек. Данных не хватает.

Выведем кинимальное и максимальное время исследования

In [ ]:
df['Время'].describe()

<ipython-input-40-a3e55bae777a>:1: FutureWarning:

Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.



count                  104420
unique                    356
top       2022-12-01 04:14:00
freq                      864
first     2022-12-01 00:39:00
last      2022-12-01 06:34:00
Name: Время, dtype: object

Вывод: данные выгрузки в датасете есть по дате за 12.01.22 года с 0:39 по 6:34.

**Исследовательский анализ данных**

Ниже рассмотрим медианные цены в аптеках по сетям в целом и ниже - отдельно по городам.

In [ ]:
aptecas = df.groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas

Сайт  Цена на сайте
0       stolichki.ru         411.00
1        maksavit.ru         390.00
2         acmespb.ru         375.00
3  planetazdorovo.ru         368.00
4      aptekiplus.ru         344.60
5          apteka.ru         321.85

Визуализируем на графике

In [ ]:
import plotly.express as px
fig = px.bar(aptecas,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

Самая дорогая сеть аптек - stolichki.ru с медианой 411 рублей, а самая дешевая - apteka.ru по всей России, с медианой около 322 р.

Ниже, по условию договора, исследуем цены в аптеках по указанным городам.

In [ ]:
aptecas_kzn = df[df['Регион'] == 'Казань'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_kzn

Сайт  Цена на сайте
0        maksavit.ru          385.0
1  planetazdorovo.ru          363.5
2      aptekiplus.ru          343.4
3          apteka.ru          329.5

In [ ]:
fig = px.bar(aptecas_kzn,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_ufa = df[df['Регион'] == 'Уфа'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_ufa

Сайт  Цена на сайте
0    maksavit.ru         367.00
1  aptekiplus.ru         344.35
2      apteka.ru         318.00

In [ ]:
fig = px.bar(aptecas_ufa,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_bgr = df[df['Регион'] == 'Белгород'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_bgr

Сайт  Цена на сайте
0  planetazdorovo.ru          398.0
1      aptekiplus.ru          347.9
2          apteka.ru          310.2

In [ ]:
fig = px.bar(aptecas_bgr,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_tul = df[df['Регион'] == 'Тула'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_tul

Сайт  Цена на сайте
0        maksavit.ru         406.50
1       stolichki.ru         396.75
2  planetazdorovo.ru         375.00
3      aptekiplus.ru         344.75
4          apteka.ru         328.30

In [ ]:
fig = px.bar(aptecas_tul,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_spb = df[df['Регион'] == 'Санкт-Петербург'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_spb

Сайт  Цена на сайте
0       stolichki.ru          420.5
1         acmespb.ru          375.0
2  planetazdorovo.ru          358.0
3          apteka.ru          315.1

In [ ]:
fig = px.bar(aptecas_spb,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_vld = df[df['Регион'] == 'Владимир'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_vld

Сайт  Цена на сайте
0       stolichki.ru         415.50
1        maksavit.ru         400.00
2      aptekiplus.ru         344.85
3  planetazdorovo.ru         344.00
4          apteka.ru         319.60

In [ ]:
fig = px.bar(aptecas_vld,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_ekb = df[df['Регион'] == 'Екатеринбург'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_ekb

Сайт  Цена на сайте
0  planetazdorovo.ru          357.0
1      aptekiplus.ru          344.5
2          apteka.ru          319.5

In [ ]:
fig = px.bar(aptecas_ekb,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_vrn = df[df['Регион'] == 'Воронеж'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_vrn

Сайт  Цена на сайте
0        maksavit.ru          412.0
1  planetazdorovo.ru          409.0
2      aptekiplus.ru          344.1
3          apteka.ru          310.8

In [ ]:
fig = px.bar(aptecas_vrn,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

In [ ]:
aptecas_ast = df[df['Регион'] == 'Астрахань'].groupby('Сайт').agg({'Цена на сайте' : 'median'}).sort_values(by='Цена на сайте', ascending=False).reset_index()
aptecas_ast

Сайт  Цена на сайте
0        maksavit.ru         353.00
1      aptekiplus.ru         346.30
2          apteka.ru         339.15
3  planetazdorovo.ru         330.00

In [ ]:
fig = px.bar(aptecas_ast,
             x='Цена на сайте',
             y='Сайт',
             text='Цена на сайте',
             color='Сайт'
            )

fig.update_layout(title='Распределение средних цен препаратов по сетям аптек',
                   xaxis_title='Средняя цена препарата',
                   yaxis_title='Сети аптек')
fig.update_xaxes(range=[300, 430])
fig.show()

# Общий вывод по городам

Не во всех городах присутствуют сети аптек, которые есть в нашей таблице.

Посчитаем, сколько всего уникальных препаратов в каждом городе. Сгруппируем по столбцу "Товар (офф. Наименование)". Выведем топ 15 из них.

In [ ]:
summ_site = df[df['Товар (офф. Наименование)'].notnull()]
sum_aptecas = summ_site['Товар (офф. Наименование)'].value_counts().reset_index().head(15)
sum_aptecas.columns = ['Товар (офф. Наименование)', 'count']
sum_aptecas

Товар (офф. Наименование)  count
0          Кеторол р-р д/инъекц. 30мг 1мл №10 амп.    139
1                           Компливит таб. п/о №60    124
2                         Римантадин таб. 50мг №20    114
3                             Трависил сироп 100мл    101
4                             Индап таб. 2.5мг №30     99
5                                         Чага 50г     87
6   Дексаметазон р-р д/инъекц. 4мг/мл 1мл №25 амп.     73
7                            Вертран таб. 24мг №30     71
8                              Глюкометр Контур ТС     70
9                      Конкор АМ таб.5мг+10 мг №30     68
10                            Гепариновая мазь 25г     66
11      Алмагель А суспензия д/приема внутрь 170мл     63
12                                      Спорыш 50г     62
13                               Энап таб. 5мг №20     62
14                        Аквалор софт спрей 150мл     62

In [ ]:
city_site = df.groupby(['Товар (офф. Наименование)', 'Регион'])['Цена на сайте'].count().reset_index()
city_site.columns = ['Товар (офф. Наименование)', 'Регион', 'count']
city_site.sort_values('count', ascending=False)

Товар (офф. Наименование)           Регион  count
17938        Кеторол р-р д/инъекц. 30мг 1мл №10 амп.             Тула     19
19323                         Компливит таб. п/о №60         Владимир     19
17933        Кеторол р-р д/инъекц. 30мг 1мл №10 амп.         Владимир     18
34767                       Римантадин таб. 50мг №20         Владимир     18
19328                         Компливит таб. п/о №60             Тула     18
...                                              ...              ...    ...
24279  Мелоксикам р-р д/инъекц.10мг/мл 1,5мл №3 амп.              Уфа      1
24278  Мелоксикам р-р д/инъекц.10мг/мл 1,5мл №3 амп.             Тула      1
24277  Мелоксикам р-р д/инъекц.10мг/мл 1,5мл №3 амп.  Санкт-Петербург      1
24276  Мелоксикам р-р д/инъекц.10мг/мл 1,5мл №3 амп.           Казань      1
24820                       Метопролол таб. 50мг №30         Белгород      1

[49640 rows x 3 columns]

In [ ]:
group_city_site = city_site[city_site['Товар (офф. Наименование)'].isin(sum_aptecas['Товар (офф. Наименование)'])]
group_city_site

Товар (офф. Наименование)           Регион  count
930    Аквалор софт спрей 150мл        Астрахань      7
931    Аквалор софт спрей 150мл         Белгород      5
932    Аквалор софт спрей 150мл         Владимир      9
933    Аквалор софт спрей 150мл          Воронеж      7
934    Аквалор софт спрей 150мл     Екатеринбург      5
...                         ...              ...    ...
48247         Энап таб. 5мг №20     Екатеринбург      7
48248         Энап таб. 5мг №20           Казань      7
48249         Энап таб. 5мг №20  Санкт-Петербург      8
48250         Энап таб. 5мг №20             Тула      9
48251         Энап таб. 5мг №20              Уфа      3

[135 rows x 3 columns]

Рассмотрим, какие уникальные значения есть в столбце Товара.

In [ ]:
df['Товар (офф. Наименование)'].nunique()

6532

Расчитаем количество уникальных товаров, (SKU), собираемых на каждом из сайтов
должно составлять не менее 1000 в день. Воспользуемся функцией группировки и выведем таблицу с количеством уникальных товаров по каждой аптеке по городам.

In [ ]:
aptecas_1000 = df.groupby(['Сайт', 'Регион']).agg({'Товар (офф. Наименование)' : 'nunique', 'Сайт' : 'unique'})['Товар (офф. Наименование)'].sort_values(ascending=False)
aptecas_1000

Сайт               Регион         
stolichki.ru       Санкт-Петербург    4648
                   Владимир           4205
maksavit.ru        Воронеж            4108
aptekiplus.ru      Астрахань          4025
                   Воронеж            4019
                   Тула               4005
                   Уфа                3997
                   Екатеринбург       3989
                   Казань             3986
                   Владимир           3984
stolichki.ru       Тула               3982
aptekiplus.ru      Белгород           3974
maksavit.ru        Тула               3787
acmespb.ru         Санкт-Петербург    3503
maksavit.ru        Владимир           2939
                   Казань             2757
                   Астрахань          2408
planetazdorovo.ru  Санкт-Петербург    2114
                   Казань             2067
                   Екатеринбург       2033
maksavit.ru        Уфа                2029
apteka.ru          Санкт-Петербург    1995
                   

Вывод: в нашем исследовании только одна аптечная сеть имеет меньше 1000 наименований товаров, - это сеть planetazdorovo.ru в Астрахани.

# Общий вывод

В этой работе мы проанализировали ценовую политику аптечных сетей по каждому региону (городу). Были выведены медианные цены на сайте по каждой сети.
Анализ был проведен на ценах, которые были указаны в столбце. Их обработка была затруднена из-за равномерного распределения (не было сильно выбивающихся значений), и отсутствия дополнительного условия касательно ценовой политики.Наиболее дорогостоящие препараты были отмечены в stolichki.ru с медианой 411 рублей, а самая дешевая - apteka.ru по всей России, с медианой около 322р. По городам, - самые дорогие аптеки – maksavit, planetazdorovo и stolichki. Дешевле препараты можно приобрести на сайтах аптек aptekiplus, acmespb и apteka.ru. По условию договора нам нужно, чтобы количество уникальных товаров в аптеках не превышало 1 тысячу. В данном случае, мы можем принять к исследованию весь датасет, кроме аптечной сети planetazdorovo в Астрахани. Там количество уникальных товаров меньше 1000 и составляет всего 762. Кроме этого, по условиях договора нам нужны данные по мониторингу цен за СУТКИ. Но данные выгружены только за первые 6 часов суток. Поэтому все расчеты велись именно по этому промежутку времени. Таким образом, выявлены несоответствия условиям договора:
1) Выгрузка цена с официального сайта произведена некорректно;
2) Мониторинг цен производился с 0:39 по 6:34 12 января 2022 года, а по условию - нужен суточный мониторинг.
3) В таблице отсутствуют несколько аптечных сайтов: aptekamos.ru, ozerki.ru, po-aptekam.ru и rigla.ru.

In [ ]:
df.to_csv('sample.csv')
from google.colab import files
files.download("sample.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>